core/commands/add.go

```diff
@@ -81,6 +81,11 @@ You can now refer to the added file in a gateway, like so:
 			return nil
 		}
 
+		_, found, _ := req.Option(progressOptionName).Bool()
```

Comment:
I have missed that the code is in PreRun function.

Everything is much less hacky now.